### ISSUE TITLE
Filter out object properties by property name filter/search

### ISSUE THREAD

**Issue by emattias** [emberjs/ember-inspector#1171]
It would be useful to filter properties on an object in the object inspector sidebar. 

Something like this:

![CleanShot 2020-04-06 at 11 41 56@2x](https://user-images.githubusercontent.com/351537/78545364-f8a0b600-77fb-11ea-8e9a-5eb20491dd56.png)

Similar to how you can filter in other places in devtools. Ex:

![CleanShot 2020-04-06 at 11 45 21](https://user-images.githubusercontent.com/351537/78545518-300f6280-77fc-11ea-9c13-2a799d447f74.gif)

**Comment by rwwagner90** [comment]
I definitely agree. This is a great feature suggestion!

**Comment by ansmonjol** [comment]
Hello @emattias,

Great idea ! 

I've started to work on that feature. The way it works for now is: 
If the user start typing a custom search, he's redirected to the "All" filter section and a filter is applied on the object properties displayed (flat).

[This MR](https://github.com/emberjs/ember-inspector/pull/1172) is still a "work in progress" and I've written what I need to improve in a dedicated section.

Please let me know if the described behaviour suit you and what you imagined.

**Comment by ansmonjol** [comment]
Clear button is now implemented, you can test by using this branch version: https://github.com/ansmonjol/ember-inspector/tree/filter-object-property

Tests are still missing.

Also questioning myself about the current behaviour.
When typing a value in the input, the user is switched to the "All" filter.
Then he's able to switch back manually to the "Grouped" filter.
It currently show the attribute list grouped but without custom filter applied (the value typed in the input is still typed.

Should we:
1. clear the input value if the user click on the "Grouped" button ?
2. Apply the custom filter on the grouped view.

WDYT ?

**Comment by nummi** [comment]
I'm guessing this would be difficult with the current implementation but I would expect the grouped view to expand the grouping if a match is found.

**Comment by ansmonjol** [comment]
Agree with you @nummi !

It's done here https://github.com/emberjs/ember-inspector/pull/1172/commits/b82218e61fcab5dbd52dabf7902871a162515db6#diff-fa24e8e76afe9d0c18545f325e0f8f4cR5
Then we do not have inconsistency between the two "All" and "Grouped" views :)

**Comment by ansmonjol** [comment]
The solution I linked above seems to break some tests, regarding computed usage.

Trying to fix, but if it takes too much time I will probably got for the first option described, and try to do the filter in "Grouped" view later.

**Comment by ansmonjol** [comment]
I've made the solution 1 from this comment https://github.com/emberjs/ember-inspector/issues/1171#issuecomment-610513112

I've tested the custom filter behaviour but I have an error while clicking on a button.
I've difficulties to find the root cause and how to fix it.
Note that it works locally if I check the `Hide container` option while running tests.
Can someone help ?

Here is the failing test (line 125 in `Run test`)
https://github.com/emberjs/ember-inspector/pull/1172/checks?check_run_id=579683560
